test/opengl/testing.h

```diff
@@ -0,0 +1,93 @@
+#ifndef _TESTING_H_
+#define _TESTING_H_
+
+#include "Halide.h"
+#include <iostream>
+#include <exception>
+#include <functional>
+#include <cmath>
+
+namespace Testing {
+
+    // fill 3-dimension buffer
+    template <typename T>
+        void fill(Halide::Buffer<T> &buf, std::function<T(int x, int y, int c)> f)
+        {
+            buf.for_each_element([&](int x, int y, int c) { 
+                    buf(x, y, c) = f(x, y, c);
+                    });
+        }
+
+    template <typename T>
+        bool neq(T a, T b, T tol)
+        {
+            return std::abs(a-b) > tol;
+        }
+
+    // check 3-dimension buffer
+    template<typename T>
+        bool check_result(const Halide::Buffer<T> &buf, std::function<T(int x, int y, int c)> f, T tol=0)
+        {
+            class err : std::exception {
+                public: static void vector(const std::vector<T> &v) {
+                    for (size_t i=0; i<v.size(); i++) {
+                        if (i > 0) std::cerr << ",";
+                        std::cerr << v[i]+0; // need to add 0 to get output -- compiler bug??
+                    }
+                }
+            };
+            try {
```

Comment:
Yes, on my platform Halide is built with exceptions disabled.  But in the Makefile that's only for building the Halide runtime; the tests don't have exceptions disabled.  And the Makefile has [[L133]](https://github.com/halide/Halide/blob/master/Makefile#L133) `CXX_FLAGS += -funwind-tables` with a comment `"on some hosts... Without -funwind-tables, libHalide.so fails to propagate exceptions and causes a test failure."` 

All of which seems to tell me it should be OK to use exceptions in the tests.  Is there some config in which the tests have exceptions disabled?